### ISSUE TITLE
Build failed on current master

### ISSUE THREAD

**Issue by ycollet** [coin-or/Cbc#276]
I tested Cbc on Visual Studio 2013 and I've got the following error during compilation:
```
Cbc_C_Interface.cpp
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1514) : error C2065: 'not'ÿ: identificateur non d‚clar‚
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1514) : error C2146: erreur de syntaxeÿ: absence de ')' avant l'identificateur 'cbc_annouced'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1514) : error C2059: erreur de syntaxeÿ: ')'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1515) : error C2144: erreur de syntaxeÿ: 'char' doit ˆtre pr‚c‚d‚ de '}'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1515) : error C2064: le terme ne correspond pas … une fonction qui prend 1 arguments
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1515) : error C2144: erreur de syntaxeÿ: 'char' doit ˆtre pr‚c‚d‚ de ';'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1541) : error C2059: erreur de syntaxeÿ: 'if'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1541) : error C2143: erreur de syntaxeÿ: absence de ';' avant '{'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1541) : error C2447: '{'ÿ: en-tˆte de fonction manquant (liste formelle … l'ancien formatÿ?)
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1551) : error C2059: erreur de syntaxeÿ: 'if'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1551) : error C2143: erreur de syntaxeÿ: absence de ';' avant '{'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1551) : error C2447: '{'ÿ: en-tˆte de fonction manquant (liste formelle … l'ancien formatÿ?)
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1619) : error C2059: erreur de syntaxeÿ: 'switch'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1619) : error C2143: erreur de syntaxeÿ: absence de ';' avant '{'
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1619) : error C2447: '{'ÿ: en-tˆte de fonction manquant (liste formelle … l'ancien formatÿ?)
F:\jenkins\workspace\Coin-Cbc\Cbc_svn\Cbc\src\Cbc_C_Interface.cpp(1640) : error C2143: erreur de syntaxeÿ: absence de ';' avant '->'
```

**Comment by tkralphs** [comment]
If you could specify the SHA of the failing commit, that would be helpful. We are currently only testing on Visual Studio versions 2015, 2017, and 2019. Everything seems fine there. Is there a specific use case for 2013? Feel free to submit a PR if it's something easy to patch (looks like).

**Comment by ycollet** [comment]
5 days ago, the code was compiling fine on Visual 2013.
If look at the error trace, the first line points to the problematic line.

Where did you anounce that CoinOR projects only focuse on Visual Studio versions 2015, 2017, and 2019 ?

----- Mail original -----
De: "Ted Ralphs" <notifications@github.com>
À: "coin-or/Cbc" <Cbc@noreply.github.com>
Cc: "ycollet" <ycollette.nospam@free.fr>, "Author" <author@noreply.github.com>
Envoyé: Lundi 9 Décembre 2019 21:14:11
Objet: Re: [coin-or/Cbc] Build failed on current master (#276)



If you could specify the SHA of the failing commit, that would be helpful. We are currently only testing on Visual Studio versions 2015, 2017, and 2019. Everything seems fine there. Is there a specific use case for 2013? Feel free to submit a PR if it's something easy to patch (looks like).

**Comment by ycollet** [comment]
So, for the fix:
Replace at line 1514:
```
  if (not cbc_annouced) {
```
by
```
  if (! cbc_annouced) {
```

**Comment by tkralphs** [comment]
Ah, OK, yes, we should avoid the use of the keyword `not`, although it has apparently been in the C++ standard since the beginning. There is actually a flag to enable it as a keyword in MSVC 2013: http://stackoverflow.com/a/555524/368896 @h-g-s Can you take care of fixing this and avoid the use of this keyword going forward?

The Windows builds we do are visible here

https://ci.appveyor.com/project/tkralphs/cbc-s1r7t 

In principle, I could add a Visual Studio 2013 build, but in the interest of not wasting too many computational resources, I thought testing three versions back was sufficient, especially since the compilers are freely available and it's easy to upgrade if an older version stops working. If you think there is a case to be made for testing MSVC 2013, let me know and we can add it.

**Comment by h-g-s** [comment]
Hi @tkralphs , trying to commit here, but it seems that now subgit is producing an error in pre-commit and is not sending the changes to svn:

Enviando       CbcMipStartIO.cpp
Enviando       Cbc_C_Interface.cpp
Transmitindo dados do arquivo ..done
Committing transaction...
svn: E165001: Commit falhou (detalhes a seguir):
svn: E165001: Commit blocked by pre-commit hook (exit code 1) with output:

SubGit ERROR REPORT (SubGit version 2.0.2-EAP ('Patrick') build #2731):

You've received this message because SubGit (http://subgit.com/) is installed in your repository
and an error that needs to be dealt with has occurred in SubGit translation engine.

TEMPORARY ERROR:
        Failed to launch background translation process: timeout waiting for pid file '/var/www/svn/Cbc/subgit/daemon.pid'.

CURRENT STATE:
        Both Git and Subversion repository are open for pushes or commits.
        Your commit was not committed, but you may retry it.

TO RECOVER:
          A) Address the problem if possible and then retry commit
        OR
          B) Run on the server
             $ subgit uninstall /var/www/svn/Cbc

        IMPORTANT: As soon as SubGit is uninstalled, both Git and Subversion repositories
                   will become open, but no synchronization will take place.

TO REPORT:
        Report an issue at http://issues.tmatesoft.com/
        You may find logs on the server at '/var/www/svn/Cbc/subgit/logs'

THANK YOU!

**Comment by tkralphs** [comment]
Fixed by 088bfbeb4e640a5567e8d55c3ed84089fd49097c.

**Comment by ycollet** [comment]
Build fine on VS 2013 ...
